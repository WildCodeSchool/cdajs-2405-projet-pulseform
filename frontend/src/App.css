@font-face {
	font-family: "Urbanist";
	src: url("./assets/fonts/Urbanist.ttf");
}

#root {
	font-family: 'Urbanist', sans-serif;
	margin: 0;
	padding: 0;
	height: 100%;
	width: 100%;
}

p {
	margin: 0;
}

*::-webkit-scrollbar {
	height: 8px;
	width: 8px;
}

*::-webkit-scrollbar-track {
	border-radius: 4px;
	background-color: #132945;
	border: 1px solid #a9a9a9;
}

*::-webkit-scrollbar-track:hover {
	background-color: #132945;
}

*::-webkit-scrollbar-track:active {
	background-color: #132945;
}

*::-webkit-scrollbar-thumb {
	border-radius: 8px;
	background-color: #d4d4d4;
}

*::-webkit-scrollbar-thumb:hover {
	background-color: #d4d4d4;
}

*::-webkit-scrollbar-thumb:active {
	background-color: #d4d4d4;
}