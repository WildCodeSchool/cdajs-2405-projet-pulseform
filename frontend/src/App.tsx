import { useQuery } from "@apollo/client";
import { ME_QUERY } from "./graphql/queries";
import "./App.css";
import LandingPage from "./pages/LandingPage";

function App() {
  const { data, loading } = useQuery(ME_QUERY);

  if (loading) return <p>Loading...</p>;

  return <LandingPage user={data?.me} />;
}

export default App;
