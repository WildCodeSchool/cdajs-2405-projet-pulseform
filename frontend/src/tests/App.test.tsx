import { render } from "@testing-library/react";
import App from "../App";

describe("App", () => {
  it("renders the App component", () => {
    const tree = render(<App />);

    expect(tree).toMatchSnapshot();
  });
});
