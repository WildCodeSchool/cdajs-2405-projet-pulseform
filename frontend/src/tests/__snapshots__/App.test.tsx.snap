// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders the App component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="landing-page"
      >
        <div
          class="landing-page__first-view landing-page__scroll"
        >
          <div
            class="landing-page__first-view__blob"
          >
            <img
              alt="blob"
              class="landing-page__first-view__blob__img"
              src="/src/assets/blobs/blob.svg"
            />
          </div>
          <div
            class="landing-page__first-view__girl"
          >
            <img
              alt="girl"
              class="landing-page__first-view__girl__img"
              src="/src/assets/images/girl.svg"
            />
          </div>
          <div
            class="landing-page__first-view__header"
          >
            <div
              class="landing-page__first-view__header__logo-m"
            >
              <div
                class="little-logo little-logo--mobile"
              >
                <img
                  alt="Logo Pulse Form"
                  class="little-logo__img"
                  src="/src/assets/icons/mobile-logo.svg"
                />
                <p
                  class="little-logo__title little-logo__title--mobile"
                >
                  Pulse Form
                </p>
              </div>
            </div>
            <div
              class="landing-page__first-view__header__logo-d"
            >
              <div
                class="little-logo little-logo--desktop"
              >
                <img
                  alt="Logo Pulse Form"
                  class="little-logo__img"
                  src="/src/assets/icons/mobile-logo.svg"
                />
                <p
                  class="little-logo__title little-logo__title--desktop"
                >
                  Pulse Form
                </p>
              </div>
            </div>
            <div
              class="landing-page__first-view__header__container"
            >
              <div
                class="landing-page__first-view__header__container__buttom"
              >
                <button
                  class="basic-button"
                  style="width: 200px; height: 50px; font-size: 18px;"
                  type="button"
                >
                  CREATE_ACCOUNT
                </button>
              </div>
              <p
                class="landing-page__first-view__header__container__connect"
              >
                CONNECT_MY
              </p>
            </div>
          </div>
          <p
            class="landing-page__first-view__description"
          >
            LANDING_DESCRIPTION
          </p>
          <div
            class="landing-page__first-view__buttom"
          >
            <button
              class="basic-button"
              style="width: 300px; height: 50px; font-size: 18px;"
              type="button"
            >
              CREATE_ACCOUNT
            </button>
          </div>
          <div
            class="landing-page__first-view__buttom-program"
          >
            <button
              class="basic-button"
              style="width: 300px; height: 80px; font-size: 24px;"
              type="button"
            >
              TEST_PROGRAM
            </button>
          </div>
        </div>
        <div
          class="landing-page__second-view landing-page__scroll"
        >
          <p
            class="landing-page__second-view__description"
          >
            LANDING_DESCRIPTION_2
          </p>
          <div
            class="landing-page__second-view__step"
          >
            <p
              class="landing-page__second-view__step__content"
            >
              PROGRAMS_TO_DICOVER
            </p>
            <p
              class="landing-page__second-view__step__content"
            >
              FOLLOW_YOUR_PROGRESS
            </p>
            <p
              class="landing-page__second-view__step__content"
            >
              PERSONALIZED_COACHING
            </p>
          </div>
          <div
            class="landing-page__second-view__buttom"
          >
            <button
              class="basic-button"
              style="width: 300px; height: 50px; font-size: 18px;"
              type="button"
            >
              TEST_PROGRAM
            </button>
          </div>
          <div
            class="landing-page__second-view__blob"
          >
            <img
              alt="blob"
              class="landing-page__second-view__blob__img"
              src="/src/assets/blobs/blob2.svg"
            />
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="landing-page"
    >
      <div
        class="landing-page__first-view landing-page__scroll"
      >
        <div
          class="landing-page__first-view__blob"
        >
          <img
            alt="blob"
            class="landing-page__first-view__blob__img"
            src="/src/assets/blobs/blob.svg"
          />
        </div>
        <div
          class="landing-page__first-view__girl"
        >
          <img
            alt="girl"
            class="landing-page__first-view__girl__img"
            src="/src/assets/images/girl.svg"
          />
        </div>
        <div
          class="landing-page__first-view__header"
        >
          <div
            class="landing-page__first-view__header__logo-m"
          >
            <div
              class="little-logo little-logo--mobile"
            >
              <img
                alt="Logo Pulse Form"
                class="little-logo__img"
                src="/src/assets/icons/mobile-logo.svg"
              />
              <p
                class="little-logo__title little-logo__title--mobile"
              >
                Pulse Form
              </p>
            </div>
          </div>
          <div
            class="landing-page__first-view__header__logo-d"
          >
            <div
              class="little-logo little-logo--desktop"
            >
              <img
                alt="Logo Pulse Form"
                class="little-logo__img"
                src="/src/assets/icons/mobile-logo.svg"
              />
              <p
                class="little-logo__title little-logo__title--desktop"
              >
                Pulse Form
              </p>
            </div>
          </div>
          <div
            class="landing-page__first-view__header__container"
          >
            <div
              class="landing-page__first-view__header__container__buttom"
            >
              <button
                class="basic-button"
                style="width: 200px; height: 50px; font-size: 18px;"
                type="button"
              >
                CREATE_ACCOUNT
              </button>
            </div>
            <p
              class="landing-page__first-view__header__container__connect"
            >
              CONNECT_MY
            </p>
          </div>
        </div>
        <p
          class="landing-page__first-view__description"
        >
          LANDING_DESCRIPTION
        </p>
        <div
          class="landing-page__first-view__buttom"
        >
          <button
            class="basic-button"
            style="width: 300px; height: 50px; font-size: 18px;"
            type="button"
          >
            CREATE_ACCOUNT
          </button>
        </div>
        <div
          class="landing-page__first-view__buttom-program"
        >
          <button
            class="basic-button"
            style="width: 300px; height: 80px; font-size: 24px;"
            type="button"
          >
            TEST_PROGRAM
          </button>
        </div>
      </div>
      <div
        class="landing-page__second-view landing-page__scroll"
      >
        <p
          class="landing-page__second-view__description"
        >
          LANDING_DESCRIPTION_2
        </p>
        <div
          class="landing-page__second-view__step"
        >
          <p
            class="landing-page__second-view__step__content"
          >
            PROGRAMS_TO_DICOVER
          </p>
          <p
            class="landing-page__second-view__step__content"
          >
            FOLLOW_YOUR_PROGRESS
          </p>
          <p
            class="landing-page__second-view__step__content"
          >
            PERSONALIZED_COACHING
          </p>
        </div>
        <div
          class="landing-page__second-view__buttom"
        >
          <button
            class="basic-button"
            style="width: 300px; height: 50px; font-size: 18px;"
            type="button"
          >
            TEST_PROGRAM
          </button>
        </div>
        <div
          class="landing-page__second-view__blob"
        >
          <img
            alt="blob"
            class="landing-page__second-view__blob__img"
            src="/src/assets/blobs/blob2.svg"
          />
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
